src/Chart/chartWidthHOC.js

```diff
@@ -15,20 +15,26 @@ export default function chartWidthHOC(BaseComponent) {
 
     state = {}
 
-    componentWillReceiveProps() {
+    componentDidMount() {
+      window.addEventListener('resize', this.handleResize)
       this.updateWidth()
     }
 
-    componentDidMount() {
-      window.addEventListener('resize', this.handleResize)
+    componentWillReceiveProps() {
       this.updateWidth()
     }
 
     componentWillUnmount() {
       window.removeEventListener('resize', this.handleResize)
     }
 
-    updateWidth = () => {
+    handleResize = throttle(this.updateWidth, 500)
```

Comment:
Either `updateWidth` needs to be a class property, or this needs to be `() => this.updateWidth()`. Otherwise, the method won't be called with the correct `this` binding.

Reply:
Done!